File: model/src/main/java/com/sreMake/model/BaseEntity.java
package com.sreMake.model;

import com.sreMake.model.user.User;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.OffsetDateTime;

@MappedSuperclass
public interface BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();


    OffsetDateTime createAt();

    @Nullable
    OffsetDateTime updateAt();

    @Nullable
    @ManyToOne
    User createBy();

    @Nullable
    @ManyToOne
    User updateBy();
}

File: model/src/main/kotlin/com/SReMake/model/BaseEntity.kt
package com.SReMake.model

import com.SReMake.model.user.User
import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

@MappedSuperclass
interface BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    val createAt: LocalDateTime
    val updateAt: LocalDateTime?

    @ManyToOne
    val createBy: User?

    @ManyToOne
    val updateBy: User?
}